source/lambda_functiontim.py

```diff
@@ -38,32 +44,46 @@ def setMetadata(formatted_tim, tim_dict):
-	formatted_tim['metadata_serialId_recordId'] = tim_dict['serialId']['recordId']
-	formatted_tim['metadata_serialId_serialNumber'] = tim_dict['serialId']['serialNumber']
-	formatted_tim['metadata_receivedAt'] = tim_dict['odeReceivedAt'].replace('Z[UTC]','')
-	formatted_tim['metadata_rmd_elevation'] = tim_dict['receivedMessageDetails']['locationData']['elevation']
-	formatted_tim['metadata_rmd_heading'] = tim_dict['receivedMessageDetails']['locationData']['heading']
-	formatted_tim['metadata_rmd_latitude'] = tim_dict['receivedMessageDetails']['locationData']['latitude']
-	formatted_tim['metadata_rmd_longitude'] = tim_dict['receivedMessageDetails']['locationData']['longitude']
-	formatted_tim['metadata_rmd_speed'] = tim_dict['receivedMessageDetails']['locationData']['speed']
-	formatted_tim['metadata_rmd_rxSource'] = tim_dict['receivedMessageDetails']['rxSource']
-	formatted_tim['metadata_schemaVersion'] = tim_dict['schemaVersion']
+	schemaVersion = tim_dict.get('schemaVersion')
+	formatted_tim['metadata_schemaVersion'] = schemaVersion
+	formatted_tim['metadata_generatedAt'] = tim_dict.get('recordGeneratedAt', '').replace('Z[UTC]','')
+	formatted_tim['metadata_recordGeneratedBy'] = tim_dict.get('recordGeneratedBy')
+	formatted_tim['metadata_sanitized'] = str(tim_dict.get('sanitized', ''))
+	formatted_tim['metadata_payloadType'] = tim_dict.get('payloadType')
+	formatted_tim['metadata_serialId_streamId'] = tim_dict.get('serialId', {}).get('streamId')
+	formatted_tim['metadata_serialId_bundleSize'] = tim_dict.get('serialId', {}).get('bundleSize')
+	formatted_tim['metadata_serialId_bundleId'] = tim_dict.get('serialId', {}).get('bundleId')
+	formatted_tim['metadata_serialId_recordId'] = tim_dict.get('serialId', {}).get('recordId')
+	formatted_tim['metadata_serialId_serialNumber'] = tim_dict.get('serialId', {}).get('serialNumber')
+	formatted_tim['metadata_receivedAt'] = tim_dict.get('odeReceivedAt', '').replace('Z[UTC]','')
+
+	# version 5
+	formatted_tim['metadata_logFileName'] = tim_dict.get('logFileName')
+	formatted_tim['metadata_recordType'] = tim_dict.get('recordType')
+	formatted_tim['metadata_rmd_elevation'] = tim_dict.get('receivedMessageDetails', {}).get('locationData', {}).get('elevation')
+	formatted_tim['metadata_rmd_heading'] = tim_dict.get('receivedMessageDetails', {}).get('locationData', {}).get('heading')
+	formatted_tim['metadata_rmd_latitude'] = tim_dict.get('receivedMessageDetails', {}).get('locationData', {}).get('latitude')
+	formatted_tim['metadata_rmd_longitude'] = tim_dict.get('receivedMessageDetails', {}).get('locationData', {}).get('longitude')
+	formatted_tim['metadata_rmd_speed'] = tim_dict.get('receivedMessageDetails', {}).get('locationData', {}).get('speed')
+	formatted_tim['metadata_rmd_rxSource'] = tim_dict.get('receivedMessageDetails', {}).get('rxSource')
+
+	# version 6
+	formatted_tim['metadata_request_rsus'] = json.dumps(tim_dict.get('request', {}).get('rsus'))
```

Comment:
I think best practice would be to call `tim_dict.get('request', {})` once, store it in a variable and use it multiple times and not rely on python optimization to be smart enough not to search for `request` over and over. Same goes with other places in the code.

Reply:
Good point! Will clean this up.